### PR TITLE
Splitting the component in smaller functional components and typescript

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Splitting the component in smaller functional components and typescript

The idea is to migrate later the `Table` component to a functional component
and typescript, but this is a good first step.

#### Ticket Link
[MM-41292](https://mattermost.atlassian.net/browse/MM-41292)

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/admin_console/jobs/table.jsx [hunk 1]
```diff
@@ -79,17 +77,10 @@ class JobTable extends React.PureComponent {
     constructor(props) {
         super(props);
         this.interval = null;
-
```
Comment (jespino) [inline components/admin_console/jobs/table.jsx:82]: I removed the loading stated because wasn't used anywhere.

**Review by devinbinnie — APPROVED** [review]

**Comment by jgilliam17** [comment]
Thanks @jespino 
Tested manually, just waiting for the E2E report before approving.
- Verified tables in the admin console are shown correctly.

**Comment by jgilliam17** [comment]
One failure on the [cypress report](https://mattermost-cypress-report.s3.amazonaws.com/108788-fc8e8b6-onprem-ent-webapp-pr-9491-mattermost/mm-ee-testfc8e8b6/mochawesome.html) 
[MM-T1201](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T1201) Remove and re-add license - Permissions freeze in place when license is removed (and then re-added)
@jespino  would this be related to your changes?

**Comment by jespino** [comment]
@jgilliam17 I'll take a look.

**Comment by jespino** [comment]
@jgilliam17 looks like it is not related, it looks like some kind of problem hitting the server API but I'm not changing anything related to that.

**Comment by jgilliam17** [comment]
Thanks for checking @jespino 
Looks good to merge.

**Review by jgilliam17 — APPROVED** [review]

**Pull request merged**